core/integer/divide_spec.rb

```diff
@@ -55,10 +71,34 @@
       (bignum_value(88) / 4294967295.5).should be_close(2147483648.25, TOLERANCE)
     end
 
+    it "returns result the same class as the argument" do
+      (@bignum / 4).should == 2305843009213693974
+      (@bignum / 4.0).should be_close(2305843009213693974, TOLERANCE)
+      (@bignum / Rational(4, 1)).should == Rational(2305843009213693974, 1)
+    end
+
     it "does NOT raise ZeroDivisionError if other is zero and is a Float" do
       (bignum_value / 0.0).to_s.should == 'Infinity'
       (bignum_value / -0.0).to_s.should == '-Infinity'
     end
+
+    ruby_version_is ""..."2.4" do
+      it "raises a TypeError if other is zero and not a Float" do
+        lambda { @bignum.send(@method, 0) }.should raise_error(TypeError, /nil is not a symbol nor a string/)
+      end
+    end
```

Comment:
again it's very strange behaviour

Reply:
This example is not in a shared spec, so `@method` is probably nil?

Reply:
oh